src/include/sof/lib/dma.h

```diff
@@ -130,6 +130,7 @@ struct dma_sg_config {
 	uint32_t dest_dev;
 	uint32_t cyclic;			/* circular buffer */
 	uint64_t period;
+	int srcid;
```

Comment:
Why do you add `srcid`? Where and for what do you use it?

I see is used in `edma.c` (line 338, above), but instead of 
```
if (srcid) {
...
} 
```
you can use an `#ifdef CONFIG_IMX8ULP`

Reply:
The reason is srcid may be a common resource, when edma really need. And this won't block existing platforms.

Reply:
Commom resource for what? What is used for? 
All I understand right now is that srcid is for 8ULP.  But I don't know what it represent - a source id for what?
If is only used to determine it's 8ULP use instead of:
```
if (srcid) {
...
} 
```
you can use an `#ifdef CONFIG_IMX8ULP`

Reply:
The name srcid is a copy name from edma driver, I think its meaning should be a source id, it represent dmamux number.